### PR TITLE
Update search for modern theme

### PR DESCRIPTION
**Pull request by johnhwhite**
[design](https://blackbaud.invisionapp.com/console/share/UG1N3XTCC7/483395718)

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box](https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box)

File: src/app/public/modules/search/search.component.html [hunk 1]
```diff
@@ -46,19 +51,16 @@
...
-              <sky-icon
-                class="sky-search-icon-default"
+              <sky-icon *skyThemeIf="'default'"
                 icon="times"
-              >
-              </sky-icon>
-              <sky-icon
-                class="sky-search-icon-modern"
+              ></sky-icon>
+              <sky-icon *skyThemeIf="'modern'"
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/search/search.component.html:57]: The "X" button looks a little large in modern theme, when compared to the design specs. Saw this in Chrome/Mac.

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Just one item, and a question: I noticed search doesn't have any accessibility tests. Since you're adding some label id referencing to the template, would you mind adding one? Thanks!

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box](https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box](https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box](https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box](https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box)

**Comment by Blackbaud-AlexKingman** [comment]
A11y tests look good 👍

**Comment by Blackbaud-AlexKingman** [comment]
Looks like the padding in modern theme is off a bit. Also... are the icon colors right?
Screen Shot 2021-04-26 at 11 17 35 AM

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box](https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box)

**Comment by Blackbaud-AdamFunderburk** [comment]
This is looking awesome. A couple points:

- The design intent was to use plain text (regular color, no italics) for the "label" but _hide it_ when the field has focus, rather than using placeholder text. That's not a hard requirement, but is it feasible?
- Even if the text remains a placeholder, the search icon should use regular text color in all states

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box](https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box)

**Comment by Blackbaud-AdamFunderburk** [comment]
The text changes look spot-on. Last thing: the border on hover when there's search text entered.
image

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box](https://developer.blackbaud.com/skyux-lookup?_sp.spa=pr-modern-search-box)

**Comment by Blackbaud-AdamFunderburk** [comment]
LGTM

**Review by Blackbaud-PaulCrowder — APPROVED** [review]

**Pull request merged**